loadtest/control/simulcontroller/actions.go

```diff
@@ -769,6 +769,52 @@ func (c *SimulController) createPost(u user.User) control.UserActionResponse {
 	return control.UserActionResponse{Info: fmt.Sprintf("post created, id %v", postId)}
 }
 
+func (c *SimulController) attachFilesToDraft(u user.User, draft *model.Draft) error {
+	type file struct {
+		data   []byte
+		upload bool
+	}
+	filenames := []string{"test_upload.png", "test_upload.jpg", "test_upload.mp4"}
+	files := make(map[string]*file, len(filenames))
+
+	for _, filename := range filenames {
+		files[filename] = &file{
+			data:   control.MustAsset(filename),
+			upload: rand.Intn(2) == 0,
+		}
+	}
+
+	// We make sure at least one file gets uploaded.
+	files[filenames[rand.Intn(len(filenames))]].upload = true
+
+	var wg sync.WaitGroup
+	fileIds := make(chan string, len(files))
+	for filename, file := range files {
+		if !file.upload {
+			continue
+		}
+		wg.Add(1)
+		go func(filename string, data []byte) {
+			defer wg.Done()
+			resp, err := u.UploadFile(data, draft.ChannelId, filename)
+			if err != nil {
+				c.status <- c.newErrorStatus(err)
+				return
+			}
+			c.status <- c.newInfoStatus(fmt.Sprintf("file uploaded, id %v", resp.FileInfos[0].Id))
+			fileIds <- resp.FileInfos[0].Id
+		}(filename, file.data)
+	}
+
+	wg.Wait()
+	numFiles := len(fileIds)
+	for i := 0; i < numFiles; i++ {
+		draft.FileIds = append(draft.FileIds, <-fileIds)
+	}
+
```

Comment:
This is a complicated way to achieve this. We can just have a slice of fileIds and let each goroutine set the fileId at separate indices of that slice. That's perfectly valid and not a race condition. And then just append that slice to `draft.FileIds`